### PR TITLE
Refactor/project packer interfaces

### PR DESCRIPTION
**Pull request by alexnm**

### CODE REVIEW

File: packages/teleport-project-packer-playground/src/index.ts [hunk 1]
```diff
@@ -152,7 +149,23 @@ export const createPlaygroundPacker = (params: PackerFactoryParams = {}) => {
     packer.setAssets(projectAssets)
     packer.setGeneratorFunction(projectGenerator.generateProject)
     packer.setPublisher(projectPublisher)
-    packer.setTemplate(projectTemplate)
+
+    const remoteGithubTemplate =
+      projectTemplates[packTechnology.type] || projectTemplates[TEMPLATES.REACT_NEXT]
+
+    if (packParams.template) {
```
Comment (alexnm) [inline packages/teleport-project-packer-playground/src/index.ts]: @ionutpasca please review this flow here

File: packages/teleport-project-packer/src/index.ts [hunk 2]
```diff
@@ -49,37 +47,24 @@ export const createProjectPacker: PackerFactory = (params: PackerFactoryParams =
...
+  ): Promise<GeneratedFolder> => {
     try {
-      template.templateFolder = await fetchTemplate(template.remote)
-      templateLoaded = true
-      return { success: true, payload: template.templateFolder }
+      template = await fetchTemplate(remoteDefinition)
+      return template // TODO: is this useful?
     } catch (err) {
-      return { success: false, payload: err }
+      throw err // TODO: error handling here?
```
Comment (ionutpasca) [inline packages/teleport-project-packer/src/index.ts]: Maybe we should only return a response similar to this

```js
{
  success: false
  payload: err
}
```

File: packages/teleport-project-packer/src/index.ts [hunk 3]
```diff
@@ -49,37 +47,24 @@ export const createProjectPacker: PackerFactory = (params: PackerFactoryParams =
...
-    }
-
+    remoteDefinition: RemoteTemplateDefinition
+  ): Promise<GeneratedFolder> => {
     try {
-      template.templateFolder = await fetchTemplate(template.remote)
-      templateLoaded = true
-      return { success: true, payload: template.templateFolder }
+      template = await fetchTemplate(remoteDefinition)
+      return template // TODO: is this useful?
```
Comment (ionutpasca) [inline packages/teleport-project-packer/src/index.ts]: There might be cases in which, after the user loads the template, wants to parse it in some way. So I think that returning the template after we are loading it is a good thing. Maybe just:

```js
return await fetchTemplate(remoteDefinition)
```

Comment (alexnm) [reply packages/teleport-project-packer/src/index.ts]: but then the template inside the packer closure would not be updated. we would have to call setTemplate after loadTemplate

**Pull request merged**